static/app/utils/profiling/hooks/useVirtualizedTree/virtualizedTreeUtils.tsx

```diff
@@ -285,7 +287,13 @@ export function computeVirtualizedTreeNodeScrollTop(
   }
 
   if (block === 'center') {
-    return newPosition - scrollHeight / 2 + rowHeight;
+    return Math.max(
```

Comment:
small nit, but maybe we should leave a small comment about how we're bounding the scrolltop so we don't get a negative or something beyond the renderable elements

Reply:
Also, wouldn't all of these theoretically need this boundary calculation?

Reply:
Yeah I _think_ the other block types are probably unbounded right now, so it might make sense to also put safeguards around them. Although, none of the other block types seem to be in use right now, so I haven't been testing those behaviours. lmk if you prefer I update those as well, or we can tackle it in the future

Reply:
Since it's only used once with this prop, I think we could just leave it for now and keep the change simple.

Reply:
@edwardgou-sentry @narsaynorath In what case does an overscroll happen? Is it due to user action or do we improperly initialize something? I cant tell exactly from the description

Reply:
https://user-images.githubusercontent.com/22846452/220958109-25f7a847-cf34-404e-abc2-96da18d40d9d.mov


https://user-images.githubusercontent.com/22846452/220958121-2c1473fa-ab1f-4c71-81fe-0b46597d94ff.mov

Here are some examples where this happens. Basically it tries to scroll the row to the center, but if there aren't enough items to satisfy that scroll, the ghost row goes to the center but the selected row does not (and the hover row is displaced by the distance between the real row and the center). It seems to work if the row isn't in view but if it is it has this behaviour. It fixes itself if you trigger a scroll though

Reply:
Hey @JonasBa ! Here's a couple scenarios where overscroll happens. It's usually when you click to an element on the flamegraph/viewhierarchy that is near the bottom of the tree. The tree tries `center` the selected node, but since the node is near the very bottom of the tree, it's not possible to fully center which causes the overscrolling effect. This can also happen when the node is at the very top of the tree too, which is why we bounded the minimum return value to 0.

https://user-images.githubusercontent.com/83961295/220958341-bce4b7d5-1296-44e7-99e0-37ecc377fcb1.mov
https://user-images.githubusercontent.com/83961295/220958353-9d6a3c2c-55a2-42e0-893e-db578e371011.mov

Reply:
haha oops, nar responded before me, but he's entirely correct too.

Reply:
Ahh, got it. Thanks for the explanation!